File: java/program/Prodi.java
import java.util.List;
import java.util.Vector;

public class Prodi {

    private String name;
    private String code;
    private Vector<Course> courses = new Vector<Course>();

    public Prodi(String name, String code){
        this.name = name;
        this.code = code;
    }

    public void setCode(String code) {
        this.code = code;
    }

    public void setName(String name) {
        this.name = name;
    }

    public void addCourse(String name, String code){
        this.courses.add(new Course(name, code));
    }

    public void removeCourse(String input){
        this.courses.removeIf(n -> (input.equalsIgnoreCase(n.getName()) || input.equals(n.getCode())));
    }

    public String getName() {
        return name;
    }

    public String getCode() {
        return code;
    }

    public List<Course> getCourses() {
        return courses;
    }
}

File: cpp/program/Prodi.cpp
#pragma once

#include <string>
#include <vector>
#include <algorithm>
#include "Course.cpp"

class Prodi{
/**
 * Class prodi memiliki atribut 
 * string name dan code. Selain itu,
 * ada vector of object address untuk
 * menampung objek Course
 * Method berupa setter dan getter
 * ada juga add dan remove untuk vector
 * of object
 */
private:

    std::string name;
    std::string code;
    std::vector<Course*> courses;

public:

    Prodi(std::string name, std::string code){
        this->name = name;
        this->code = code;
    }

    ~Prodi(){};

    void setName(std::string name){
        this->name = name;
    }

    void setCode(std::string code){
        this->code = code;
    }

    void addCourse(std::string name, std::string code){
        this->courses.push_back(new Course(name, code));
    }

    void removeCourse(std::string input){
        auto i = std::find_if(this->courses.begin(), this->courses.end(), [&input](Course* temp){return (temp->getName() == input) || (temp->getCode() == input);});
        if (i != this->courses.end()) this->courses.erase(i);
    }

    std::string getName(){
        return this->name;
    }

    std::string getCode(){
        return this->code;
    }

    const std::vector<Course*>& getCourses() {
        return this->courses;
    }

};